Add configuration section for OpenIDConnect plugin to Dashboard docs (#181)
mirumee/saleor-docs: docs/dashboard/configuration/plugins.md

Below you can find an example configuration of the Adyen plugin:

![Adyen](../screenshots/config-plugins-adyen.png)

## OpenID Connect

### Enabling OpenID Connect plugin

> Go to Configuration -> Plugins -> OpenID Connect and fill in the fields:

Reviewer: ```suggestion
Go to Configuration -> Plugins -> OpenID Connect and fill in the following fields:
```

- **Client ID:** Your client ID, required to authenticate on the OAuth provider side.
- **Client secret:** Your client secret, required to authenticate on the OAuth provider side.
- **Enable refresh token:** Determine if the refresh token should be also fetched from the provider. By disabling it, users will need to re-login after the access token expired. By enabling it, frontend apps will be able to refresh the access token.
- **OAuth authorization url:** Based on the authorization URL, Saleor will generate the redirect URL for the authorization request.
- **OAuth token url:** The URL used to exchange received OAuth code to the OAuth token.
- **JSON web key set url:** The JSON Web Key Set (JWKS) is a set of keys containing the public 
keys used to verify any JSON Web Token (JWT) issued by the authorization server.

> - **OAuth logout url:** The logout URL which Saleor will return for frontend's logout request.

Reviewer 2: "url" -> "URL"

- **Audience:** The Oauth resource identifier. If provided, Saleor will define the audience for each authorization request.  Used to fetch user permissions from OAuth provider and map it to Saleor's permission.
- **Use OAuth scope permissions:** Use OAuth scope permissions to grant a logged-in user access to protected resources. Your OAuth provider needs to have defined Saleor's permission scopes in format _saleor:<saleor-permission\>.

:::note

> After the user finishes the authentication process, the OAuth provider redirects the user to Saleor to exchange code for the token. Make sure that your OAuth provider has whitelisted Saleor's redirect URL.

Reviewer: ```suggestion
Once the user completes the authentication process, the OAuth provider redirects the user to Saleor to exchange code for a token. Make sure that your OAuth provider has whitelisted Saleor's redirect URL.
```

> `http(s)://<your-saleor-backend>/plugins/mirumee.authentication.openidconnect/callback`

Reviewer 2: Should there be a trailing space here? If not, why not?

Reviewer 2: Also: nitpick but I think we should only ever mention `https` as the protocol.

Author: yes, you are right. I will change this.

Author: Which trailing space are you talking about?

Reviewer 2: Sorry, that was a typo, trailing *slash*.

Author: It will work with and without trailing slash. The plugin receives the internal URL path as str provided by PluginManager. If it starts with `/callback`, it tries to handle it as a callback action. 
Do you want me, to add a trailing slash to the example url?

Reviewer 2: I think there should only be one working address so if we can enforce it being `/callback/`, I'd prefer that. If we allow random suffixes to work, we will have to keep supporting that for the rest of our lives.

:::

Below you can find an example configuration of the OpenID Connect plugin:
![OpenID Connect](../screenshots/config-plugins-openid-connect.png)

### Using OAuth permissions in Saloer

> If `Use OAuth scope permissions` is enabled, Saleor will request his own permissions as OAuth scopes.  Each permission has the prefix `saleor:`.  If the user has assigned Saleor's permissions on the OAuth side, Saleor will grant them to the logged-in user.

Reviewer: ```suggestion
When the `Use OAuth scope permissions` is enabled, Saleor requests its own permissions as OAuth scopes. Each permission has the prefix `saleor:`. If the users assign Saleor permissions on the OAuth side, then Saleor will grant them to the logged-in user.
```
